pkg/model/bootstrapscript.go

```diff
@@ -133,6 +142,24 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup, cs *kops.Cluste
 	return fi.WrapResource(templateResource), nil
 }
 
+// getRelevantHooks returns a list of hooks to be applied to the instance group
+func (b *BootstrapScript) getRelevantHooks(hooks []kops.HookSpec, role kops.InstanceGroupRole) []kops.HookSpec {
+	relevantHooks := []kops.HookSpec{}
+	for _, hook := range hooks {
+		if len(hook.Roles) == 0 {
```

Comment:
We pass the full list of hook specs and roles can be defined differently for each hook, so will need to check on each one.